var/spack/repos/builtin/packages/warpx/package.py

```diff
@@ -68,7 +68,7 @@ class Warpx(CMakePackage):
     depends_on('ascent +mpi', when='+ascent +mpi')
     depends_on('blaspp', when='+psatd dims=rz')
     depends_on('blaspp +cuda', when='+psatd dims=rz compute=cuda')
-    depends_on('boost@1.66.0: +math', when='+qedtablegen')
+    depends_on('boost@1.66.0: +math cxxstd=14', when='+qedtablegen')
```

Comment:
Now I wonder, since we require C++14 or newer, so a compiler might actually pick C++17 or 2a.
Is there some automatism in the boost package for this?

Reply:
Nope, you will have to copy the variant to your own package and do the depends_on('x cxxstd=...', when='cxxstd=...') dance for all values you allow

Reply:
oh my. let's just keep `cxxstd=14` here for now :D It's a non-default variant and we only use some math parts.